rust/chains/hyperlane-cosmos/src/mailbox/contract.rs

```diff
@@ -0,0 +1,246 @@
+use std::fmt::{Debug, Formatter};
+use std::num::NonZeroU64;
+use std::str::FromStr;
+
+use async_trait::async_trait;
+use cosmrs::proto::cosmos::base::abci::v1beta1::TxResponse;
+use tracing::{instrument, warn};
+
+use hyperlane_core::{
+    utils::bytes_to_hex, ChainResult, ContractLocator, HyperlaneChain, HyperlaneContract,
+    HyperlaneDomain, HyperlaneMessage, HyperlaneProvider, Mailbox, RawHyperlaneMessage,
+    TxCostEstimate, TxOutcome, H256, U256,
+};
+
+use crate::address::CosmosAddress;
+use crate::grpc::WasmProvider;
+use crate::payloads::general;
+use crate::payloads::mailbox::{
+    GeneralMailboxQuery, ProcessMessageRequest, ProcessMessageRequestInner,
+};
+use crate::types::tx_response_to_outcome;
+use crate::utils::get_block_height_for_lag;
+use crate::{payloads, ConnectionConf, CosmosProvider, Signer};
+
+#[derive(Clone)]
+/// A reference to a Mailbox contract on some Cosmos chain
+pub struct CosmosMailbox {
+    config: ConnectionConf,
+    domain: HyperlaneDomain,
+    address: H256,
+    provider: CosmosProvider,
+}
+
+impl CosmosMailbox {
+    /// Create a reference to a mailbox at a specific Ethereum address on some
```

Comment:
nit: I know this was already like this but a drive-by fix
```suggestion
    /// Create a reference to a mailbox at a specific Cosmos address on some
```

Reply:
Done